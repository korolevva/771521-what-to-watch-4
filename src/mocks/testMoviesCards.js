export const moviesCards = [
  {
    id: 1,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    imagePreview: `img/bohemian-rhapsody.jpg`,
    poster: `https://placeimg.com/270/410/arch/grayscale`,
    title: `Bohemian Rhapsody`,
    genre: `Biography`,
    date: 2018,
    description: `The story of the legendary British rock band Queen and lead singer Freddie Mercury`,
    rating: 8.0,
    ratingCount: 305,
    director: `Bryan Singer`,
    stars: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
    duration: 167,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 2,
    background: `https://placeimg.com/1300/552/tech`,
    imagePreview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    poster: `https://placeimg.com/270/410/arch/grayscale`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    genre: `Drama`,
    date: 2018,
    description: `The second installment of the "Fantastic Beasts" series featuring the adventures of Magizoologist Newt Scamander.`,
    rating: 6.6,
    ratingCount: 423,
    director: `David Yates`,
    stars: [`Eddie Redmayne`, `Katherine Waterston`, `Dan Fogler`],
    duration: 90,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
];
