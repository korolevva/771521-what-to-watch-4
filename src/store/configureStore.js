import {createStore, compose, applyMiddleware} from "redux";
import {rootReducer} from "../reducers/index.js";
import {createAPI} from "../api.js";
import thunk from "redux-thunk";
import {Operation as DataOperation} from "../actions/dataActions.js";
import {Operation as UserOperation} from "../actions/userActions.js";
import {requireAuthorization} from "../actions/userActions.js";
import {AuthorizationStatus} from "../reducers/user/user.js";

const onUnauthorized = () => {
  store.dispatch(requireAuthorization(AuthorizationStatus.NO_AUTH));
};

const api = createAPI(onUnauthorized);

export const store = createStore(
    rootReducer,
    compose(
        applyMiddleware(thunk.withExtraArgument(api)),
        window.__REDUX_DEVTOOLS_EXTENSION__ ? window.__REDUX_DEVTOOLS_EXTENSION__() : (f) => f
    )
);

store.dispatch(DataOperation.loadMovie());
store.dispatch(DataOperation.loadMovies());
store.dispatch(UserOperation.checkAuth());

