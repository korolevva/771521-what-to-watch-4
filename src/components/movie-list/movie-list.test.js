import React from "react";
import renderer from "react-test-renderer";
import MovieList from "./movie-list.jsx";
import {moviesCards} from "../../mocks/testMoviesCards.js";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import NameSpace from "../../reducers/name-space.js";
import {Router} from "react-router-dom";
import history from "../../history.js";

const mockStore = configureStore([]);
const store = mockStore({
  [NameSpace.GENRE]: {currentGenre: `All genre`},
  [NameSpace.DATA]: {moviesCards},
  [NameSpace.MOVIE_CARD]: {displayedMoviesCards: 8}
});

it(`Render MovieList`, () => {
  const tree = renderer
    .create(<Provider store={store}>
      <Router
        history={history}
      >
        <MovieList
          moviesCards={moviesCards}
          displayedMoviesCards={8}
        />
      </Router>
    </Provider>,
    {
      createNodeMock: () => {
        return {};
      }
    }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
