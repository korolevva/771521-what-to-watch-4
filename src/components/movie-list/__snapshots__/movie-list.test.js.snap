// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MovieList 1`] = `
Array [
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        autoPlay={false}
        controls={false}
        height="175"
        muted={true}
        poster="img/bohemian-rhapsody.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          type="video/mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        autoPlay={false}
        controls={false}
        height="175"
        muted={true}
        poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          type="video/mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        autoPlay={false}
        controls={false}
        height="175"
        muted={true}
        poster="img/macbeth.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          type="video/mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Macbeth
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        autoPlay={false}
        controls={false}
        height="175"
        muted={true}
        poster="img/aviator.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          type="video/mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Aviator
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        autoPlay={false}
        controls={false}
        height="175"
        muted={true}
        poster="img/we-need-to-talk-about-kevin.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          type="video/mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        We need to talk about Kevin
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        autoPlay={false}
        controls={false}
        height="175"
        muted={true}
        poster="img/what-we-do-in-the-shadows.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          type="video/mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        What We Do in the Shadows
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        autoPlay={false}
        controls={false}
        height="175"
        muted={true}
        poster="img/revenant.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          type="video/mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Revenant
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        autoPlay={false}
        controls={false}
        height="175"
        muted={true}
        poster="img/johnny-english.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          type="video/mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Johnny English
      </a>
    </h3>
  </article>,
]
`;
